client/Lykke.AlgoStore.MatchingEngineAdapter.Client/MeaCommunicator.cs

```diff
@@ -0,0 +1,94 @@
+﻿using Common.Log;
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Sockets;
+using System.Threading;
+using Lykke.AlgoStore.MatchingEngineAdapter.Abstractions.Domain;
+using Lykke.AlgoStore.MatchingEngineAdapter.Abstractions.Domain.Listening.Requests;
+using Lykke.AlgoStore.MatchingEngineAdapter.Abstractions.Domain.Listening.Responses;
+using Lykke.AlgoStore.MatchingEngineAdapter.Abstractions.Services.Listening;
+
+namespace Lykke.AlgoStore.MatchingEngineAdapter.Client
+{
+    internal class MeaCommunicator : IMeaCommunicator
+    {
+        private static readonly Dictionary<byte, Type> _defaultMessageTypeMap = new Dictionary<byte, Type>
+        {
+            [(byte)MeaResponseType.Pong] = typeof(PingRequest),
+            [(byte)MeaResponseType.MarketOrderResponse] = typeof(ResponseModel<double>)
+        };
+
+        private TcpClient _tcpClient = new TcpClient();
```

Comment:
Don't forget disposing disposable

Reply:
ok, we will check again what we've missed